src/script/common/c_content.cpp

```diff
@@ -593,6 +593,12 @@ ContentFeatures read_content_features(lua_State *L, int index)
 	// Set to true if wall_mounted used to be set to true
 	getboolfield(L, index, "legacy_wallmounted", f.legacy_wallmounted);
 
+	// Client shall immediately place this node when player digs the node.
+	// Server will update the precise end result a moment later.
+	// "" = no prediction
+	getstringfield(L, index, "node_dig_prediction",
+			f.node_dig_prediction);
+
```

Comment:
yeah, I rewrote this part:

"" - will disable dig prediction
"air" - the default value - will predict to remove the node
any other value will attempt to place the node as named in the string